# 📊 Muestreo Aleatorio Simple (MAS) en Python

Este notebook presenta dos ejemplos prácticos del diseño de Muestreo Aleatorio Simple (MAS),
una técnica fundamental en estadística que consiste en seleccionar una muestra de manera que
todas las posibles combinaciones de tamaño n tengan la misma probabilidad de ser elegidas.

🎯 Objetivos:
1. Simular poblaciones reales con características distintas.
2. Aplicar muestreo aleatorio simple usando pandas.
3. Calcular estimaciones de parámetros poblacionales y su error estándar.

📌 Ejemplos:
- Ejemplo 1: Encuesta de satisfacción estudiantil en una universidad.
- Ejemplo 2: Estimación del gasto promedio en una tienda online.

Para cada caso, se calcula la media muestral y su error estándar usando la fórmula:

    se(𝑦̄) = sqrt( (1 - f) * s² / n )

donde:
- f = fracción de muestreo (n/N)
- s² = varianza muestral
- n = tamaño de la muestra
"""


In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Simulación de dos ejemplos con MAS

# Ejemplo 1: Satisfacción estudiantil
np.random.seed(42)
N1 = 5000
poblacion1 = pd.DataFrame({
    'id': np.arange(1, N1+1),
    'satisfaccion': np.random.normal(loc=7, scale=1.5, size=N1).clip(1, 10)
})
n1 = 200
muestra1 = poblacion1.sample(n=n1, replace=False)
media1 = muestra1['satisfaccion'].mean()
var1 = muestra1['satisfaccion'].var(ddof=1)
f1 = n1 / N1
se1 = np.sqrt((1 - f1) * var1 / n1)

# Ejemplo 2: Ventas en tienda online
np.random.seed(0)
N2 = 10000
poblacion2 = pd.DataFrame({
    'id': np.arange(1, N2+1),
    'monto': np.random.exponential(scale=50, size=N2)
})
n2 = 500
muestra2 = poblacion2.sample(n=n2, replace=False)
media2 = muestra2['monto'].mean()
var2 = muestra2['monto'].var(ddof=1)
f2 = n2 / N2
se2 = np.sqrt((1 - f2) * var2 / n2)

# Combinar resultados
resultados = pd.DataFrame({
    'Ejemplo': ['Satisfacción Estudiantil', 'Ventas en Tienda Online'],
    'Tamaño población': [N1, N2],
    'Tamaño muestra': [n1, n2],
    'Media muestral': [round(media1, 2), round(media2, 2)],
    'Error estándar': [round(se1, 3), round(se2, 2)]
})
print("Resultados de Muestreo Aleatorio Simple")
display(resultados)

Resultados de Muestreo Aleatorio Simple


Ejemplo  Tamaño población  Tamaño muestra  Media muestral  \
0  Satisfacción Estudiantil              5000             200            6.94   
1   Ventas en Tienda Online             10000             500           47.08   

   Error estándar  
0           0.093  
1           2.070

# 📈 Interpretación de Resultados


A partir de los resultados del muestreo aleatorio simple aplicado en ambos ejemplos, observamos:

🔹 Ejemplo 1: Satisfacción Estudiantil
- La media muestral estimada fue cercana a 6.94 sobre una escala de 1 a 10.
- El error estándar fue de aproximadamente 0.093, lo que indica **alta precisión** en la estimación
  debido a la baja variabilidad en las respuestas y un tamaño muestral relativamente adecuado.

🔹 Ejemplo 2: Ventas en Tienda Online
- El gasto promedio estimado fue de aproximadamente $47.08.
- El error estándar fue mayor (~2.07), lo cual refleja la **alta dispersión** típica de datos financieros,
  especialmente cuando siguen distribuciones asimétricas como la exponencial (muchas compras pequeñas y
  pocas muy grandes).

📌 Conclusión:
El Muestreo Aleatorio Simple permite estimar parámetros poblacionales de forma confiable, siempre que se
cumpla el supuesto de aleatoriedad y representatividad. Sin embargo, **la precisión depende del tamaño de
muestra y la variabilidad de los datos**, lo cual se refleja directamente en el error estándar.

